src/helper.cc

```diff
@@ -1159,16 +1159,16 @@ helperStatefulHandleRead(const Comm::ConnectionPointer &conn, char *, size_t len
 
     srv->roffset += len;
     srv->rbuf[srv->roffset] = '\0';
-    Helper::Xaction *r = srv->requests.front();
```

Comment:
UB aside, what about stateless helpers? Should their processes be killed if they speak too soon, for similar fundamental reasons? Keeping the two helper session classes in sync would help unify them in the future.

If you agree, please enlarge this PR scope to cover both helper types.